languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -887,6 +887,108 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
             </rule>
         </rulegroup>
 
+        <rulegroup id="CONCORDANCIA_DAR_HORAS" name="Concordância de construções como 'deram dez horas' => 'deu dez horas'" default="temp_off">
+            <rule> <!-- deu 10 horas -->
+                <pattern>
+                    <token postag="SENT_START"/> <!-- keep it tightly scoped -->
+                    <token postag_regexp="yes" postag="R." min="0" max="2"/>
+                    <marker>
+                        <token postag_regexp="yes" postag="V...3S." inflected="yes" regexp="yes">dar|bater</token>
+                    </marker>
+                    <token postag_regexp="yes" postag="Z.+"/>
+                    <token regexp="yes">horas|h(rs?)?</token> <!-- account for incorrect abbreviations as well -->
```

Comment:
It is technically incorrect, but perhaps XX:XX should be accounted for as well?

Reply:
Did it in the other sub-rule – the tokeniser keeps `hh:mm` as a single token.